sdks/java/core/src/main/java/org/apache/beam/sdk/util/WindowedValue.java

```diff
@@ -175,6 +176,18 @@ public T getValue() {
   public abstract Collection<? extends BoundedWindow> getWindows();
 
   /**
+   * Returns a collection of {@link WindowedValue WindowedValues} identical to this one, except each
+   * is in exactly one of the windows that this {@link WindowedValue} is in.
+   */
+  public Collection<WindowedValue<T>> explodeWindows() {
```

Comment:
I can use an Iterable. Done

I think using Iterables.transform is less readable than the imperative style.